### ISSUE TITLE
double URL encoding when converting JSON-LD (and N-Quads) to TTL

### ISSUE THREAD

**Issue by fennibay** [linkeddata/rdflib.js#623]
### Problem description

When converting JSON-LD (application/ld+json) file to TTL (text/turtle), URLs are encoded again, so the typical `%2520` issue appears in the TTL files.

The issue also happens when I convert JSON-LD first to N-Quads (application/n-quads) with jsonld library, and then to TTL. So I'd suspect the root cause might be in the TTL serializer.

### Example

Example input:

```json
{
    "@context": {
        "ex": "http://example.com#"
    },
    "@id": "ex:myid",
    "ex:prop1": {
        "ex:prop2%20dogan": {
            "ex:prop3": "value"
        }
    }
}
```

Example output:

```ttl
@prefix ex: <http://example.com#>.

ex:myid ex:prop1 [ <http://example.com#prop2%2520dogan> [ ex:prop3 "value" ] ].
```

`http://example.com#prop2%2520dogan` should have been `http://example.com#prop2%20dogan`. This is for instance the behavior of another implementation [EasyRdf Converter](https://www.easyrdf.org/converter).

### References

Similar issue: #290 // I'm not sure why it was closed.

**Comment by fennibay** [comment]
I found out that the issue happens when base is set to undefined in the serialize call. I created PR #625 to fix.

**Comment by angelo-v** [comment]
I assume this is fixed with the merge of #625
